CMakeLists.txt

```diff
@@ -434,92 +434,69 @@ elseif(MIOPEN_BACKEND STREQUAL "OpenCL")
     set(CPACK_RPM_PACKAGE_CONFLICTS miopen-hip)
 endif()
 
-# Unpack DB files
+set(KERNELS_BINARY_DIR ${PROJECT_BINARY_DIR}/src/kernels)
+set(KERNELS_SOURCE_DIR ${PROJECT_SOURCE_DIR}/src/kernels)
+
+find_program(UNZIPPER lbzcat bzcat REQUIRED)
+file(MAKE_DIRECTORY ${KERNELS_BINARY_DIR})
+
 function(unpack_db db_bzip2_file)
-    set(KERNELS_DIR "${CMAKE_SOURCE_DIR}/src/kernels")
-    STRING(REPLACE ".bz2" "" db_file "${db_bzip2_file}")
-    find_program(UNZIPPER NAMES lbunzip2 bunzip2)
-    if(EXISTS "${db_file}")
-        message(STATUS "WARNING: ${db_file} already exists and will be overwritten")
-    endif()
-    if(EXISTS "${db_bzip2_file}")
-        message(STATUS "${UNZIPPER} -k -f ${db_bzip2_file}")
-        execute_process(
-            COMMAND ${UNZIPPER} -k -f ${db_bzip2_file}
-            WORKING_DIRECTORY ${KERNELS_DIR}
-            RESULT_VARIABLE ret
-            )
-        if(NOT ret EQUAL "0")
-            message(FATAL_ERROR "${db_file} could not be extracted, file is empty!")
-            execute_process(
-                COMMAND touch ${db_file}
-                WORKING_DIRECTORY ${KERNELS_DIR}
-                )
-        endif()
-    endif()
+    get_filename_component(__fname ${db_bzip2_file} NAME_WLE)
+    add_custom_command(OUTPUT ${KERNELS_BINARY_DIR}/${__fname}
+                       COMMAND ${UNZIPPER} -k ${db_bzip2_file} > ${KERNELS_BINARY_DIR}/${__fname})
+    string(REPLACE "." "_" __tname ${__fname})
```

Comment:
If we unzip in the build directory we should unzip in the same relative path to the `so` file so that MIOpen can find the file at runtime.

Reply:
i.e. extract to `<build_dir>/share/miopen/db/`